src/Diagram/stateManager/nodeManager/index.ts

```diff
@@ -217,6 +233,53 @@ export class NodeManager {
       this.eventBus.publish(Events.DiagramEvents.RenderRequested);
     }
   };
+  foldUnfoldChildren = () => {
+    const { node, io } = this.state.hover;
+    if (node && !io) {
+      const root = this.state.nodes.find(f => f.id === node.id);
+      const childrenToFold = new Array<Node>();
+      let foundChildren = new Array<Node>();
+      let foundAllChildren = false;
+
+      if (root === undefined) {
+        return;
+      }
+      let newParents = [root];
+      while (!foundAllChildren) {
+        foundChildren = this.findChildren(newParents);
+        newParents = foundChildren;
+        childrenToFold.push(...foundChildren);
+        foundAllChildren = foundChildren.length === 0 ? true : foundAllChildren;
```

Comment:
you can rewrite it as `if (foundChildren.length < 1) { break; }` and not use `foundAllChildren` at all

Reply:
Corrected